pkg/ccl/changefeedccl/sink.go

```diff
@@ -249,20 +250,28 @@ func getSink(
 			return makeNullSink(&changefeedbase.SinkURL{URL: u}, metricsBuilder(nullIsAccounted))
 		case isKafkaSink(u):
 			return validateOptionsAndMakeSink(changefeedbase.KafkaValidOptions, func() (Sink, error) {
+				targets, err := AllTargets(ctx, feedCfg, serverCfg.ExecutorConfig.(*sql.ExecutorConfig))
```

Comment:
You should be able to pass the targets into this function instead of needing to recompute them here (same for the other sinks in this function)